scripts/consensus-release-to-chap.sh

```diff
@@ -6,45 +6,76 @@
-cardano_last_version=$(grep "<a id=" ouroboros-consensus-cardano/CHANGELOG.md  | cut -d\' -f2 | cut -d- -f2 | head -n1)
+function pkgs_in_bundle {
+  sed '/Changelog entries/q' $1/CHANGELOG.md | grep "\- \`ouroboros-" | cut -d\` -f2 | sort | uniq
+}
 
-consensus_packages=$(find . -maxdepth 1 -type d -name "ouroboros-consensus*" -not -name "*byron*" -not -name "*shelley*" -not -name "*cardano*")
-consensus_last_version=$(grep "<a id=" ouroboros-consensus/CHANGELOG.md  | cut -d\' -f2 | cut -d- -f2 | head -n1)
+function last_version {
+  ch_version=$(grep "<a id=" $1/CHANGELOG.md  | cut -d\' -f2 | cut -d- -f2 | head -n1)
+  if [[ -z $ch_version ]]; then
+      if [[ $1 == "ouroboros-consensus-test" ]]; then
+          echo "0.3.1.0"
+      elif [[ $1 == "ouroboros-consensus-cardano-test" ]]; then
+          echo "0.4.0.0"
+      fi
+  else
+      echo $ch_version
+  fi
+}
 
-if [[ ! $(git tag -l "release-consensus-$consensus_last_version") || ! $(git tag -l "release-cardano-$cardano_last_version") ]]; then
+consensus_packages=$(pkgs_in_bundle ouroboros-consensus)
+consensus_last_version=$(last_version ouroboros-consensus)
+consensus_test_packages=$(pkgs_in_bundle ouroboros-consensus-test)
+consensus_test_last_version=$(last_version ouroboros-consensus-test)
+cardano_packages=$(pkgs_in_bundle ouroboros-consensus-cardano)
+cardano_last_version=$(last_version ouroboros-consensus-cardano)
+cardano_test_packages=$(pkgs_in_bundle ouroboros-consensus-cardano-test)
+cardano_test_last_version=$(last_version ouroboros-consensus-cardano-test)
+
+if [[ ! $(git tag -l "release-consensus-$consensus_last_version") || ! $(git tag -l "release-consensus-test-$consensus_test_last_version") || ! $(git tag -l "release-cardano-$cardano_last_version") || ! $(git tag -l "release-cardano-test-$cardano_test_last_version") ]]; then
     echo "There are no tags pushed for the versions! Checkout the commit after bors merged the release PR and run ./scripts/consensus-push-tags.sh"
     exit 1
 fi
 
-git clone git@github.com:input-output-hk/cardano-haskell-packages chap
+if [[ ! -d chap ]]; then
+  git clone git@github.com:input-output-hk/cardano-haskell-packages chap
+else
+  ( cd chap
+    git checkout master
+    git pull
+  )
+fi
 
 ( cd chap
-  if [[ ! -d _sources/ouroboros-consensus/$consensus_last_version && ! -d _sources/ouroboros-consensus-cardano/$cardano_last_version ]]; then
-      branch="release-consensus-$consensus_last_version-cardano-$cardano_last_version"
-  elif [[ ! -d _sources/ouroboros-consensus/$consensus_last_version ]]; then
-      branch="release-consensus-$consensus_last_version"
-  elif [[ ! -d _sources/ouroboros-consensus/$consensus_last_version ]]; then
-      branch="release-cardano-$cardano_last_version"
-  else
-      exit 1
-  fi
 
-  git checkout -b $branch
+  git checkout -b release-consensus
```

Comment:
Don't we want to make a different branch for each release?

Reply:
Well, for CHaP I think it is good enough if we just open a branch, do the release and delete it. We don't really need a specific branch name. Although probably I need to `delete; checkout master; create` instead of only creating it